### PR TITLE
Go proc eventd

### PR DESCRIPTION
**Pull request by prashant-sb**
List and watch the Linux processes.
- List
```
$ ./run -list
 Threads     1
 Name        kworker/0
 Pid         3043
 PPid        2
 Tgid        3043
 State       I (idle)
 Umask       0000
 
 PPid        2
 Tgid        357
 State       S (sleeping)
 Umask       0000
 Threads     1
 Name        loop7
 Pid         357
 
 Tgid        62
 State       I (idle)
 Umask       0000
 Threads     1
 Name        kworker/2
 Pid         62
 PPid        2
 
 Threads     3
 Name        rtkit-daemon
 Pid         953
 PPid        1
 Tgid        953
 State       S (sleeping)
 Umask       0777
 
...
```
- Watch
```
# ./run -logtostderr -watch 2040
I0930 14:39:38.618608    2483 event_handler.go:50] Watching pid: 2040

I0930 14:39:45.949696    2483 event_handler.go:55] Fork event:{2040 2497}
I0930 14:39:45.950555    2483 event_handler.go:57] Exec event:{2497}
I0930 14:39:45.951949    2483 event_handler.go:55] Fork event:{2497 2499}
I0930 14:39:45.952262    2483 event_handler.go:57] Exec event:{2499}
I0930 14:39:45.952672    2483 event_handler.go:55] Fork event:{2499 2500}
I0930 14:39:45.952859    2483 event_handler.go:57] Exec event:{2500}
I0930 14:39:45.953221    2483 event_handler.go:59] Exit event:{2500}
I0930 14:39:45.953564    2483 event_handler.go:55] Fork event:{2499 2501}
I0930 14:39:45.953682    2483 event_handler.go:55] Fork event:{2501 2502}
I0930 14:39:45.954482    2483 event_handler.go:57] Exec event:{2502}
I0930 14:39:45.954810    2483 event_handler.go:59] Exit event:{2502}
I0930 14:39:45.954930    2483 event_handler.go:59] Exit event:{2501}
I0930 14:39:45.955025    2483 event_handler.go:59] Exit event:{2499}
I0930 14:39:45.955576    2483 event_handler.go:55] Fork event:{2497 2503}
I0930 14:39:45.955995    2483 event_handler.go:57] Exec event:{2503}
I0930 14:39:45.956423    2483 event_handler.go:59] Exit event:{2503}
I0930 14:40:01.449159    2483 event_handler.go:59] Exit event:{2040}
I0930 14:40:16.286155    2483 event_handler.go:59] Exit event:{2497}

```

### CODE REVIEW

File: proc_eventd/nlproc.go [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+	pr.Data.Id.Val = _CN_VAL_PROC
+
+	pr.Data.Len = uint16(binary.Size(op))
+
+	buf := bytes.NewBuffer(make([]byte, 0, pr.Header.Len))
+	binary.Write(buf, byteOrder, pr)
+	binary.Write(buf, byteOrder, op)
+
+	return syscall.Sendto(listener.sock, buf.Bytes(), 0, listener.addr)
+}
```
Comment (praju-1) [inline proc_eventd/nlproc.go]: Add new line at end of the file

File: proc_eventd/notifier.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+// true - in which case the caller should break from the readEvents loop.
+func (w *Watcher) isDone() bool {
+	var done bool
+	select {
+	case done = <-w.done:
+		w.finish()
+	default:
+	}
+	return done
+}
```
Comment (praju-1) [inline proc_eventd/notifier.go]: same

**Review by praju-1 — CHANGES_REQUESTED** [review]

File: proc_eventd/notifier.go (hunk 2 above)
Comment (prashant-sb) [reply proc_eventd/notifier.go]: Added

File: proc_eventd/nlproc.go (hunk 1 above)
Comment (prashant-sb) [reply proc_eventd/nlproc.go]: Done.

**Review by praju-1 — APPROVED** [review]

**Pull request merged**